File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
answer = 0

def dfs(numbers, target, index, result):
    global answer
    if index == len(numbers):
        if result == target:
            answer += 1
        return
    
    for i in [-1, +1]:
        dfs(numbers, target, index+1, result + i * numbers[index])

def solution(numbers, target):
    dfs(numbers, target, 0, 0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int answer;
int tmp[2] = {-1, 1};

void dfs(vector<int> &numbers, int &target, int index, int result) {
    if (index >= numbers.size()) {
        if (result == target) answer++;
        return;
    }
    
    for (int i : tmp) {
        dfs(numbers, target, index+1, result+i*numbers[index]);
    }
}

int solution(vector<int> numbers, int target) {
    dfs(numbers, target, 0 , 0);
    return answer;
}